x-pack/plugins/enterprise_search/public/applications/__mocks__/wait_for_promises.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * If you created an immediately resolved promise, you can use this to wait for it to resolve without a reference.
+ *
+ * http.get.mockReturnValueOnce(Promise.resolve({ analyticsUnavailable: true }));
+ * AnalyticsLogic.actions.loadQueryData('some-query');
+ * 
+ * await waitForPromises();
+        
+ * expect(AnalyticsLogic.actions.onAnalyticsUnavailable).toHaveBeenCalled();
+ *
+ */
+export function waitForPromises() {
+  let promiseResolve: (value: unknown) => void;
```

Comment:
I'm a little confused, are we using `value` here or referencing it any point in the future? Why include it if so?

```suggestion
  let promiseResolve: () => void;
```

^ does the above work as-is?

Definitely tell me if I'm wrong here, I'm not super following this helper to be honest and it seems incredibly magical to me 😅